src/main/java/com/teragrep/rlo_06/StructuredData.java

```diff
@@ -0,0 +1,268 @@
+
+                                            if (!stream.next()) {
+                                                throw new ParseException("TOO SHORT");
+                                            }
+                                            b = stream.get();
+                                        }
+                                    }
+                                } else {
+                                    elementValue.put(b);
+                                    sdElemVal_max_left--;
+
+                                    if (!stream.next()) {
+                                        throw new ParseException("TOO SHORT");
+                                    }
+                                    b = stream.get();
+                                }
+                            }
+                        } else {
+                            // skip through, no subscription for this sdElem
+
+                            if (!stream.next()) {
+                                throw new ParseException("TOO SHORT");
+                            }
+                            b = stream.get();
+                            while (b != 34) { // '"'
+                                // escaped are special: \" \\ \] ...
+                                if (b == 92) { // \
+                                    // read next
+
+                                    if (!stream.next()) {
+                                        throw new ParseException("TOO SHORT");
+                                    }
+                                    b = stream.get();
+                                    // if it is a '"' then it must be taken care of, loop can do the rest
+                                    if (b == 34) {
+
+                                        if (!stream.next()) {
+                                            throw new ParseException("TOO SHORT");
+                                        }
+                                        b = stream.get();
+                                    }
+                                } else {
+
+                                    if (!stream.next()) {
+                                        throw new ParseException("TOO SHORT");
+                                    }
+                                    b = stream.get();
+                                }
+                            }
+                        }
+
+                        // clean up sdElementIterator for the next one
+                        resultset.sdElementIterator.flip();
+                        resultset.sdElementIterator.clear();
+
+                        // take next one for the while to check if ' ' or if to break it
+
+                        if (!stream.next()) {
+                            throw new ParseException("TOO SHORT");
+                        }
+                        b = stream.get();
+                    } // while (b == 32) { // multiple ' ' separated sdKey="sdValue" pairs may exist
```

Comment:
What is this

Reply:
Note: Rather use END OF <something something> BLOCK to distinguish it from commented-out code if this is just "end of while block"